raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -281,25 +288,29 @@ library NettingChannelLibrary {
         bytes32 h;
         bytes32 hashlock;
 
-        (expiration, amount, hashlock) = decodeLock(locked_encoded);
+        // Check if caller_address is a participant and select her partner (for
+        // third party unlock see #541)
+        uint8 index = 1 - index_or_throw(self, caller_address);
```

Comment:
I don't really like this. I realize that there's not much else to do, but it just doesn't seem very intuitive when I read the code.